pkg/engine/task/task_toggle.go

```diff
@@ -0,0 +1,65 @@
+package task
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+)
+
+const (
+	toggleTaskError = "ToggleTaskError"
+)
+
+// ToggleTask will apply or delete a set of given resources to the cluster based on value of Parameter. See Run method for more details.
+type ToggleTask struct {
+	Name      string
+	Parameter string
+	Resources []string
+}
+
+func (tt ToggleTask) Run(ctx Context) (bool, error) {
+	// 1. - Get the task to run
+	task, err := tt.intermediateTask(ctx)
+	if err != nil {
+		return false, fatalExecutionError(err, toggleTaskError, ctx.Meta)
+	}
+	// 2. - Run the returned task
+	return task.Run(ctx)
+}
+
+func (tt ToggleTask) intermediateTaskSpec() v1beta1.TaskSpec {
+	return v1beta1.TaskSpec{
+		ResourceTaskSpec: v1beta1.ResourceTaskSpec{
+			Resources: tt.Resources,
+		},
+	}
+}
+
+func (tt ToggleTask) intermediateTask(ctx Context) (Tasker, error) {
+	var task Tasker
+	// 1. - Get the parameter value
+	val, exists := ctx.Parameters[tt.Parameter]
+	if !exists {
```

Comment:
This is an interesting edge case...  I would suggest that if the parameter doesn't exist that it is false and we should log a warning and continue.

Reply:
I disagree, I think bailing out loud and clear is better than to assume a default value. Default values should be set in the `params.yaml` and not in the code. 
If we default here, this could easily hide a typo when setting the parameter.

Reply:
> I would suggest that if the parameter doesn't exist that it is false

I couldn't disagree more. Fail loud and proud. Do not assume default values.